test/test/jfr/JfrTests.java

```diff
@@ -78,6 +80,91 @@ public void parseMultiModeRecording(TestProcess p) throws Exception {
+            }
+        }
+
+        assert events.contains("jdk.ExecutionSample"); // cpu profiling
+        assert events.contains("jdk.JavaMonitorEnter"); // lock profiling
+        assert events.contains("jdk.ObjectAllocationInNewTLAB"); // alloc profiling
+        assert events.contains("profiler.WallClockSample"); // wall clock profiling
+        assert events.contains("profiler.LiveObject"); // profiling of live objects
+        assert events.contains("profiler.Malloc"); // nativemem profiling
+        assert events.contains("profiler.Free"); // nativemem profiling
+    }
+
+    /**
+     * Test to validate profiling output with "--all" flag for Linux with event override
+     *
+     * @param p The test process to profile with.
+     * @throws Exception Any exception thrown during profiling JFR output parsing.
+     */
+    @Test(mainClass = JfrMutliModeProfiling.class, agentArgs = "start,all,event=java.util.Properties.getProperty,alloc=2m,file=%f.jfr", os = Os.LINUX)
+    public void allModeRecordingLinuxWithEventOverride(TestProcess p) throws Exception {
+        p.waitForExit();
+        Set<String> events = new HashSet<>();
+        try (RecordingFile recordingFile = new RecordingFile(p.getFile("%f").toPath())) {
+            while (recordingFile.hasMoreEvents()) {
+                RecordedEvent event = recordingFile.readEvent();
+                String eventName = event.getEventType().getName();
+                events.add(eventName);
+                if (eventName.equals("jdk.ExecutionSample")) {
+                    // This means that only instrumented method was profiled and overall CPU profiling was skipped
+                    assert event.getStackTrace().toString().contains("java.util.Properties.getProperty");
+                }
+            }
+        }
+        assert events.contains("jdk.JavaMonitorEnter"); // lock profiling
+        assert events.contains("jdk.ObjectAllocationInNewTLAB"); // alloc profiling
+        assert events.contains("profiler.WallClockSample"); // wall clock profiling
+        assert events.contains("profiler.LiveObject"); // profiling of live objects
+        assert events.contains("profiler.Malloc"); // nativemem profiling
+        assert events.contains("profiler.Free"); // nativemem profiling
+    }
+
+    /**
+     * Test to validate profiling output with "--all" flag for MacOS arm64
+     * Note: This can be enabled for x64 macOS after: https://github.com/async-profiler/async-profiler/issues/1193
+     *
+     * @param p The test process to profile with.
+     * @throws Exception Any exception thrown during profiling JFR output parsing.
+     */
+    @Test(mainClass = JfrMutliModeProfiling.class, agentArgs = "start,all,file=%f.jfr", os = Os.MACOS, arch = Arch.ARM64)
+    @Test(mainClass = JfrMutliModeProfiling.class, agentArgs = "start,all,lock=10ms,file=%f.jfr", os = Os.MACOS, arch = Arch.ARM64)
+    public void allModeRecordingMacOs(TestProcess p) throws Exception {
+        p.waitForExit();
```

Comment:
```suggestion
        Assert.isEqual(p.exitCode(), 0);
```
so we fail fast and we don't waste time looking into the profiling output to debug the failure. Same for other tests

Reply:
[done](https://github.com/async-profiler/async-profiler/pull/1281/commits/fa641036e7813f9fae37c8bfd04592dc18b5dae2)